src/renderer/components/Editor/TakeComponent.tsx

```diff
@@ -111,68 +121,71 @@ const TakeComponent = ({
     setIsFirstTimeOpen,
   ]);
 
+  const NumberedButton = (
+    <Avatar
+      onClick={onSelectTake}
+      sx={{
+        height: 20,
+        width: 20,
+        fontSize: 12,
+        color: colors.grey[700],
+        backgroundColor: isActive ? colors.yellow[500] : colors.grey[400],
+        display: 'flex',
+        position: 'absolute',
+        left: '-30px',
+        transform: 'translateY(2px)',
+        cursor: 'pointer',
+      }}
+    >
+      {takeIndex + 1}
+    </Avatar>
+  );
+
+  const TakeWords = takeWords.map((word, index, words) => {
+    const wordIndex = transcriptionIndex + index;
+    return (
+      <WordOuterComponent
+        key={`word-outer-${word.originalIndex}-${word.pasteKey}`}
+        word={word}
+        prevWord={wordIndex > 0 ? words[wordIndex - 1] : null}
+        nextWord={wordIndex < words.length - 1 ? words[wordIndex + 1] : null}
+        index={wordIndex}
+        isPlaying={nowPlayingWordIndex === wordIndex}
+        isPrevWordSelected={isIndexInRange(selection, wordIndex - 1)}
+        isSelected={isIndexInRange(selection, wordIndex)}
+        isNextWordSelected={isIndexInRange(selection, wordIndex + 1)}
+        onMouseDown={onWordMouseDown}
+        onMouseEnter={onWordMouseEnter}
+        isInInactiveTake={!(isActive || isTakeGroupOpened) && !isFirstTimeOpen}
+        transcriptionLength={words.length}
+        {...passThroughProps}
+      />
+    );
+  });
+
+  useLayoutEffect(() => {
+    if (takeRef && takeRef.current?.clientHeight) {
+      setCurrentTakeHeight(takeRef.current.clientHeight);
+    }
+  }, [takeRef.current?.clientHeight]);
+
   return (
     <>
       <TakeWrapper className="take" onClick={onClick}>
-        <CustomRowStack sx={{ justifyContent: 'flex-start' }}>
+        <CustomRowStack
+          sx={{ justifyContent: 'flex-start', paddingLeft: '4px', gap: '4px' }}
+        >
           {isTakeGroupOpened || isActive ? (
             <>
-              {!isFirstTimeOpen && isTakeGroupOpened && (
-                <Avatar
-                  onClick={onSelectTake}
-                  sx={{
-                    height: 22,
-                    width: 22,
-                    fontSize: 12,
-                    color: colors.grey[700],
-                    backgroundColor: isActive
-                      ? colors.yellow[500]
-                      : colors.grey[400],
-                    display: 'flex',
-                    position: 'absolute',
-                    left: '-30px',
-                    transform: 'translateY(2px)',
-                    cursor: 'pointer',
-                  }}
-                >
-                  {takeIndex + 1}
-                </Avatar>
-              )}
+              {!isFirstTimeOpen && isTakeGroupOpened && NumberedButton}
```

Comment:
This is a bit hard to read - maybe put the conditional in the declaration for `numberedButton`. I.e.

```
const numberedButton = (!isFirstTimeOpen && isTakeGroupOpened) ? (... stuff ...) : null;
```

And then just render it unconditionally.

Also `numberedButton` should start with a lowercase letter - the only things that should be in `PascalCase` are class names and component declarations (e.g. you declare a component `Thing` which can be built as `<Thing />`, this is not what `NumberedButton` is as we can't go `<NumberedButton />`)

Reply:
Yeah I wasn't sure about these conventions. Thanks for clearing them up.